tests/common.py

```diff
@@ -1646,3 +1648,47 @@ def extract_stack_to_frame(extract_stack: list[Mock]) -> FrameType:
         current_frame = next_frame
 
     return top_frame
+
+
+def setup_test_component_platform(
+    hass: HomeAssistant,
+    domain: str,
+    entities: Sequence[Entity],
+    from_config_entry: bool = False,
+) -> MockPlatform:
+    """Mock a test component platform for tests."""
+
+    async def _async_setup_platform(
+        hass: HomeAssistant,
+        config: ConfigType,
+        async_add_entities: AddEntitiesCallback,
+        discovery_info: DiscoveryInfoType | None = None,
+    ) -> None:
+        """Set up a test component platform."""
+        async_add_entities(entities)
+
+    platform = MockPlatform(
+        async_setup_platform=_async_setup_platform,
+    )
+
+    # avoid loading config_entry if not needed
+    if from_config_entry:
+        from homeassistant.config_entries import ConfigEntry
```

Comment:
We already import the `ConfigFlow` from config entries. I don't think avoiding this import will save much.

Reply:
Agreed, we don't need a late import for this

Reply:
Changed in https://github.com/home-assistant/core/pull/114260